src/text_render.rs

```diff
@@ -358,6 +372,9 @@ impl TextRenderer {
 
             self.vertex_buffer = buffer;
             self.vertex_buffer_size = buffer_size;
+
+            self.staging_belt.finish();
+            self.staging_belt = StagingBelt::new(buffer_size);
```

Comment:
As far as I can see, we only need to worry about resizing the belt when the vertex buffer grows, since index buffer uploads will never be bigger than the vertex ones.

6 `u32` indices (6 * 4 bytes) are generated per each 4 `GlyphToRender` vertices (4 * 18 bytes).